src/main/java/com/denizenscript/clientizen/objects/ItemTag.java

```diff
@@ -0,0 +1,146 @@
+    // For example, 'i@stick'.
+    //
+    // @description
+    // An ItemTag represents a holdable item generically.
+    //
+    // ItemTags are temporary objects, to actually modify an item in an inventory you must add the item into that inventory.
+    //
+    // ItemTags do NOT remember where they came from. If you read an item from an inventory, changing it
+    // does not change the original item in the original inventory. You must set it back in.
+    //
+    // Find a list of valid materials at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html>
+    // Note that some materials on that list are exclusively for use with blocks, and cannot be held as items.
+    //
+    // -->
+
+    final ItemStack itemStack;
+    public String script; // Compact with server-side item scripts
+
+    public ItemTag(ItemStack itemStack) {
+        this.itemStack = itemStack;
+    }
+
+    public ItemTag(ItemConvertible convertible) {
+        this(convertible.asItem().getDefaultStack());
+    }
+
+    @Fetchable("i")
+    public static ItemTag valueOf(String string, TagContext context) {
+        if (string.startsWith("i@")) {
+            string = string.substring("i@".length());
+        }
+        if (ObjectFetcher.isObjectWithProperties(string)) {
+            return ObjectFetcher.getObjectFromWithProperties(ClientizenObjectRegistry.TYPE_ITEM, string, context);
+        }
+        MaterialTag material = MaterialTag.valueOf(string, context);
+        if (material == null) {
+            if (context == null || context.showErrors()) {
+                Debug.echoError("valueOf ItemTag returning null: invalid item type '" + string + "' specified.");
+            }
+            return null;
+        }
+        return new ItemTag(material.state != null ? material.state.getBlock() : material.item);
+    }
+
+    public static boolean matches(String string) {
+        if (string.startsWith("i@")) {
+            return true;
+        }
+        Identifier identifier = Identifier.tryParse(string);
+        return identifier != null && (Registries.ITEM.containsId(identifier) || Registries.BLOCK.containsId(identifier));
+    }
+
+    public static void register() {
+        PropertyParser.registerPropertyTagHandlers(ItemTag.class, tagProcessor);
+    }
+
+    public static final ObjectTagProcessor<ItemTag> tagProcessor = new ObjectTagProcessor<>();
+
+    @Override
+    public ObjectTag getObjectAttribute(Attribute attribute) {
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
+
+    @Override
+    public void adjust(Mechanism mechanism) {
+        tagProcessor.processMechanism(this, mechanism);
+    }
+
+    @Override
+    public void applyProperty(Mechanism mechanism) {
+        adjust(mechanism);
+    }
+
+    public ItemStack getStack() {
+        return itemStack;
+    }
+
+    public String getName() {
+        return Utilities.idToString(Registries.ITEM.getId(itemStack.getItem()));
```

Comment:
`Identifier`s for content do not currently (and probably will never be able to) change at runtime. Registry lookups aren't exactly expensive, but we could probably have the `Identifier` as a field stored in `MaterialTag`.

Reply:
I'd feel better if it was actually a `RegistryKey<Item>` or `RegistryKey<Block>` field but that's useless generic handling for no added benefit.

Reply:
Add an `identifier` field to `ItemTag`